src/Cake.Prca/IssueFilterer.cs

```diff
@@ -0,0 +1,178 @@
+            issues.NotNull(nameof(issues));
+            issueComments.NotNull(nameof(issueComments));
+
+            this.log.Verbose("Filtering issues before posting...");
+
+            // ReSharper disable once PossibleMultipleEnumeration
+            var codeAnalysisIssues = issues as IList<ICodeAnalysisIssue> ?? issues.ToList();
+
+            var result = this.FilterIssuesByPath(codeAnalysisIssues);
+            result = this.FilterPreExistingComments(result, issueComments);
+            result = this.FilterIssuesByNumber(result);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Checks if there's already a comment for an issue.
+        /// </summary>
+        /// <param name="issue">Issue to check.</param>
+        /// <param name="issueComments">List of existing comments.</param>
+        /// <returns>True if there are already comments for an issue.</returns>
+        private static bool IssueHasMatchingComments(
+            ICodeAnalysisIssue issue,
+            IDictionary<ICodeAnalysisIssue, IEnumerable<IPrcaDiscussionComment>> issueComments)
+        {
+            return
+                issueComments.ContainsKey(issue) &&
+                issueComments[issue].Any();
+        }
+
+        /// <summary>
+        /// Filters all issues affecting files which do not belong to files changed in this pull request.
+        /// </summary>
+        /// <param name="issues">List of issues which should be filtered.</param>
+        /// <returns>List of issues filtered to only the ones affecting files changed in this pull request.</returns>
+        private IList<ICodeAnalysisIssue> FilterIssuesByPath(IList<ICodeAnalysisIssue> issues)
+        {
+            if (!issues.Any())
+            {
+                return issues;
+            }
+
+            var modifiedFilesInPullRequest = this.pullRequestSystem.GetModifiedFilesInPullRequest().ToList();
+            this.log.Verbose(
+                "Files changed in this pull request:\n{0}",
+                string.Join(
+                    Environment.NewLine,
+                    modifiedFilesInPullRequest.Select(x => "  " + x)));
+
+            var countBefore = issues.Count;
+            var result =
+                issues
+                    .Where(issue => modifiedFilesInPullRequest.Select(x => x.ToString()).Contains(issue.AffectedFileRelativePath.ToString()))
+                    .ToList();
+            var commentsFiltered = countBefore - result.Count;
+
+            this.log.Information(
+                "{0} issue(s) were filtered because they do not belong to files that were changed in this pull request",
+                commentsFiltered);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Filters issues for which already a comment exists.
+        /// Issues are filtered if the they fulfill all of the following conditions:
+        /// * A comment with the same content exists
```

Comment:
I moved the documentation of the logic to `Orchestrator.GetMatchingComments` where it actually is implemented.